seonshine_frontend/src/hooks/useTable.ts

```diff
@@ -0,0 +1,63 @@
+import { ChangeEvent, useState } from 'react';
+
+import { SortingState } from '@tanstack/react-table';
+
+import { IPlainObject } from '@/types/common';
+
+export type UseTableType = {
+  currentPage: number;
+  sorting: SortingState;
+  pageSize: number;
+  sortKey: string;
+  sortType: 'desc' | 'asc';
+  searchQuery: string;
+  searchField: string;
+  handlePageChange: (page: number) => void;
+  handleSortingChange: (newSorting: SortingState) => void;
+  handlePageSizeChange: (pageSize: number) => void;
+  handleSearchChange: (field: string, query: string) => void;
+};
+
+const useTable = (initPageSize?: number, initSearchField?: string, initSearchQuery?: string): UseTableType => {
```

Comment:
params must be an object and should be defined new type. Missing `currentPage, sortKey, sortType`